src/jsbsim_bridge.cpp

```diff
@@ -64,6 +64,7 @@ JSBSimBridge::JSBSimBridge(JSBSim::FGFDMExec *fdmexec, std::string &path)
   // Instantiate sensors
   if (CheckConfigElement(config, "sensors", "imu")) {
     _imu_sensor = std::make_unique<SensorImuPlugin>(_fdmexec);
+    _imu_sensor->setSensorConfigs(config);
```

Comment:
Here you should pass the `imu` node of the xml tree only. This way:
- IMU can only access what it needs to
- you can change the xml tree structure easily
- you can have multiple instances of a sensor

The same goes for the rest of the sensors/nodes.